File: fabric-frontend/src/app/api/models/system-description-dto.ts
/* tslint:disable */
/* eslint-disable */
import { EnvironmentDto } from './environment-dto';
import { SystemComponentDto } from './system-component-dto';
export interface SystemDescriptionDto {
  environments?: Array<EnvironmentDto>;
  settings?: { [key: string]: {  } };
  systemComponents?: Array<SystemComponentDto>;
}

File: fabric-backend-core/src/main/java/io/storydoc/fabric/systemdescription/app/SystemDescriptionDTO.java
package io.storydoc.fabric.systemdescription.app;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SystemDescriptionDTO {
    private List<SystemComponentDTO> systemComponents;
    private List<EnvironmentDTO> environments;
    private Map<String, Map<String, Map<String, String>>> settings;
}
